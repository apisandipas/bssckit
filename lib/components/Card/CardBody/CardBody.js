"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _base = require("../../Utilities/base");

var _themeFunctions = require("../../../utils/themeFunctions");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  flex: 1 1 auto;\n  padding: ", ";\n  & > a {\n    &:hover {\n      text-decoration: none;\n    }\n    & + a {\n      margin-left: ", ";\n    }\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var CardBody = (0, _styledComponents["default"])(_base.Div)(_templateObject(), function (props) {
  return (0, _themeFunctions.getPadding)(props, 'cardBody', 'default');
}, function (props) {
  return (0, _themeFunctions.getMargin)(props, 'cardBody', 'leftA');
});
var _default = CardBody;
exports["default"] = _default;