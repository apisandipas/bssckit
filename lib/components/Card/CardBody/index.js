"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "CardBody", {
  enumerable: true,
  get: function get() {
    return _CardBody["default"];
  }
});

var _CardBody = _interopRequireDefault(require("./CardBody"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }