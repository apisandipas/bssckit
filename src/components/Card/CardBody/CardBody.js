import styled from 'styled-components';
import { theme } from 'styled-tools';
import defaultTheme from '../../../theme';
import { Div } from '../../Utilities/base';

const CardBody = styled(Div)`
  flex: 1 1 auto;
  padding: ${theme('cardBody.padding.default')};
  & > a {
    &:hover {
      text-decoration: none;
    }
    & + a {
      margin-left: ${theme('cardBody.margin.leftA')};
    }
  }
`;

CardBody.defaultProps = {
  theme: defaultTheme,
};

export default CardBody;
