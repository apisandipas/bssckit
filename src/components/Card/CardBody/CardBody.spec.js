import React from 'react';
import { axe } from 'jest-axe';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import CardBody from './CardBody';

describe('CardBody Component', () => {
  it('matches CardBody', async () => {
    const { container } = render(<CardBody />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
    expect(container.firstChild).toMatchSnapshot();
  });
});
