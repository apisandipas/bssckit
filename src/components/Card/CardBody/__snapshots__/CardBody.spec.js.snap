// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardBody Component matches CardBody 1`] = `
.c0 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  padding: 1.25rem;
}

.c0 > a:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 > a + a {
  margin-left: 1.25rem;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<div
  class="sc-fznyAO c0"
/>
`;
